src/moonlink_datafusion/src/connection_pool.rs

```diff
@@ -0,0 +1,361 @@
+use std::{
+    collections::HashMap,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        LazyLock,
+    },
+    time::{Duration, Instant},
+};
+use tokio::{net::UnixStream, sync::Mutex, time::sleep};
+
+struct PooledEntry {
+    stream: UnixStream,
+    last_used: Instant,
+}
+
+static POOL: LazyLock<Mutex<HashMap<String, Mutex<Vec<PooledEntry>>>>> =
+    LazyLock::new(|| Mutex::new(HashMap::new()));
+
+const MAX_PER_URI: usize = 30;
+
+static IDLE_TIMEOUT_MS: AtomicU64 = AtomicU64::new(30_000);
```

Comment:
I only see it updated in testing, which we should inject via connection pool configs, instead of global variables.